core/chain-configs/src/genesis_config.rs

```diff
@@ -77,7 +79,7 @@ fn default_max_kickout_stake_threshold() -> u8 {
 }
 
 #[derive(Debug, Clone, SmartDefault, serde::Serialize, serde::Deserialize)]
-pub struct GenesisConfig {
+pub struct GenesisConfigSnapshot {
```

Comment:
Personally, I don't see some great use for this approach, but since you think it will leave the codebase in a better shape I will implement it that way.